### DISCUSSION TITLE
Hardware based button for display brightness

### DISCUSSION THREAD

**Discussion by faragher** [markqvist/Reticulum#519] [Ideas]
While discussing the screen and the issues with it (power consumption, light polution, burn-in, etc.) I realized you can use the serial connection to disable/dim the display, but there's no physical button. Does anyone have any thoughts on if it's important enough to set aside a GPIO (preferably an interrupt enabled one) for hardware control over the display? Just something simple like a three mode "off/on/dim" setting that's stepped through.

Seeing as how custom hardware is getting more popular, this might be a useful firmware option.

**Comment by markqvist** [comment]
A few boards have a physical button that can be used for this. I recently added a button input driver to the RNode firmware, that can differentiate between presses, long-presses, etc. It's currently only used for putting the board into deep-sleep and waking it again, but could easily be used for turning on and off the display with a short tap.

One thing to note, though, is that the two boards I've tested this on (LilyGO T3S3 and Heltec v3) have a rather strange setup of using a strangely low resistance resistor to sink the button GPIO when connected, which actually results in that part of the board heating up significantly when the button is held, and power usage increasing dramatically while it is. In normal usage, it's not really likely to be a big issue, but it's a strange design choice indeed. Just slap a 100K on there man, it's not meant to be a finger-heater.
